modules/core/src/main/java/org/apache/synapse/core/axis2/Axis2SynapseEnvironment.java

```diff
@@ -365,6 +384,14 @@ public boolean injectInbound(final MessageContext synCtx, SequenceMediator seq,
         
         // Following code is reached if the sequential==true or inbound is
         // reached max level
+        if (synCtx.getProperty("inbound.endpoint.name") != null) {
```

Comment:
Maybe it's a good idea to use an appropriate constant upon the property name "inbound.endpoint.name".